packages/headless/doc-parser/src/actions-resolver.ts

```diff
@@ -0,0 +1,225 @@
+import {emitAsTsDoc} from './tsdoc-emitter';
+
+interface ActionsNamespace {
+  name: string;
+  members: Action[];
+}
+
+enum ActionKind {
+  Normal = 'normalAction',
+  Analytics = 'analyticsAction',
+}
+
+interface BaseAction {
+  name: string;
+  desc: string;
+  parameter: AnyEntity | null;
+  actionKind: ActionKind;
+}
+
+enum AnalyticsKind {
+  Search = 'search',
+  Click = 'click',
+  Custom = 'custom',
+}
+
+interface AnalyticsAction extends BaseAction {
+  actionKind: ActionKind.Analytics;
+  analyticsKind: AnalyticsKind;
+}
+
+interface NormalAction extends BaseAction {
+  actionKind: ActionKind.Normal;
+}
+
+type Action = NormalAction | AnalyticsAction;
+
+export function resolveActionNamespace(
+  entry: ApiEntryPoint,
+  name: string
+): ActionsNamespace {
+  const namespace = findApi(entry, name) as ApiNamespace;
+  return {
+    name: namespace.displayName,
+    members: namespace.members
+      .filter((member) => member.kind === ApiItemKind.Variable)
+      .map((action) => resolveAction(entry, action as ApiVariable))
+      .filter((action) => action) as Action[],
+  };
+}
+
+function resolveAction(
+  entry: ApiEntryPoint,
+  variable: ApiVariable
+): Action | null {
+  const tokens = variable.excerpt.tokens.flatMap(({text}) =>
+    tokenizePrimitives(text)
+  );
+  if (tokens[1] === 'AsyncThunk') {
+    const parameterTypeName = getAsyncThunkParameterType(tokens);
+
+    return {
+      actionKind: ActionKind.Normal,
+      name: variable.displayName,
+      desc: variable.tsdocComment
+        ? emitAsTsDoc(
+            (variable.tsdocComment.summarySection
+              .nodes as unknown) as readonly DocNode[]
+          )
+        : '',
+      parameter:
+        parameterTypeName !== 'void'
+          ? resolveActionParameter(
+              entry,
+              parameterTypeName,
+              (variable.tsdocComment?.params?.blocks[0] ??
+                null) as DocParamBlock | null
+            )
+          : null,
+    };
+  }
+
+  if (
+    tokens[4] === 'AsyncThunkAction' &&
+    tokens[6].startsWith('AnalyticsType.')
+  ) {
+    const parameterTypeName = getAnalyticsActionParameterType(tokens);
+
+    return {
+      actionKind: ActionKind.Analytics,
+      name: variable.displayName,
+      desc: variable.tsdocComment
+        ? emitAsTsDoc(
+            (variable.tsdocComment.summarySection
+              .nodes as unknown) as readonly DocNode[]
+          )
+        : '',
+      parameter:
+        parameterTypeName !== 'void'
+          ? resolveActionParameter(
+              entry,
+              parameterTypeName,
+              (variable.tsdocComment?.params?.blocks[0] ??
+                null) as DocParamBlock | null
+            )
+          : null,
+      analyticsKind: getAnalyticsActionKind(tokens),
+    };
+  }
+
+  console.warn(
+    `Could not resolve action "${variable.displayName}" of type "${tokens
+      .slice(1)
+      .join('')}"`
+  );
```

Comment:
This currently warns on `buildSearchRequest`. A next step, along with resolution of normal actions, is going to be to resolve helpers.